doc/docs/master/deploy-manage/manage/s3gateway/deploy-s3gateway-sidecar.md

```diff
@@ -0,0 +1,92 @@
+# Create an S3-enabled Pipeline
+
+Pachyderm can deploy the S3 gateway either together with the `pachd` pod or
+as a sidecar next to your pipeline worker pod. The former is
+typically used when you need to configure an ingress or egress with
+object storage tooling, such as MinIO, boto3, and others. The latter
+is needed when you use Pachyderm with external data processing
+platforms that interact with object stores but do not work with local
+file systems, such as Kubeflow or Apache Spark.
+
+Platforms like Kubeflow run their own set of pods. If you enable an
+S3 gateway next to `pachd`, these pods connect to this *master* S3
+gateway and read files from there. The master S3 gateway lives
+independently and outside of the pipeline lifecycle. Therefore, if
+Kubeflow connects through the master S3 gateway, the Pachyderm pipelines
+created in Kubeflow do not properly maintain data provenance. When the
+S3 functionality is exposed through a sidecar instance in the
+pipeline worker pod, Kubeflow can access the files stored in S3 buckets
+in the pipeline pod, which ensures the provenance is maintained
+correctly. The S3 gateway sidecar instance is created together with the
+pipeline and shut down when the pipeline is destroyed.
+
+## Limitations
+
+Pipelines exposed through a sidecar S3 gateway have the following limitations:
+
+* Because of PFS limitations, you cannot write to the input repos exposed
+through the S3 gateway instance. Similarly, you cannot read files from
+the output bucket that you enabled by using the `s3_out` parameter in the
+pipeline spec. This limitation guarantees that pipeline provenance is
+preserved.
+
+* The `glob` field in the pipeline spec should not be specified. The glob
+pattern is automatically set to `\` and all files are processed as a single
+datum. In this configuration, already processed datums are not skipped which
+could be a performance consideration for some deployments.
+
+* Join and union inputs are not supported, but you can create union or
+specify an individual input.
+
+While you cannot create a join, you can create a union of an S3-enabled
+input with a non-S3 input. For a non-S3 input you can still specify a
+glob pattern.
+
+## Expose a Pipeline through an S3 Gateway in a Sidecar
+
+When you work with platforms like Kubeflow or Apache Spark, you need
+to spin up an S3 gateway instance that runs alongside the pipeline worker
+pod as a sidecar container. To do so, set the `s3` parameter in the `input`
+part of your pipeline specification to `true`. When enabled, this parameter
+mounts S3 buckets for input repositories in the S3 gateway sidecar instance
+instead of in `/pfs/`. You can set this property for each PFS input in
+a pipeline. The address of the input repository will be `s3://<input_repo>`.
```

Comment:
> will be `s3://<input_repo>`.

Just double checking this. In normal S3G its branch.repo -- in this one it's just repo? I assume the change is just because the branch isn't needed since there no ability to take two branches in the same repo as inputs?

Reply:
@ysimonson can you please help answer the questions above?

Reply:
@JoeyZwicker you can take two branches of the same repo as inputs fine. But every input has a name, and we use that as the bucket name.